index.js

```diff
@@ -85,263 +86,251 @@ class LatticeKeyring extends EventEmitter {
+  async signPersonalMessage(address, msg) {
     return this.signMessage(address, { payload: msg, protocol: 'signPersonal' });
   }
 
-  signTypedData(address, msg, opts) {
-    if (opts.version && (opts.version !== 'V4' && opts.version !== 'V3'))
-      throw new Error(`Only signTypedData V3 and V4 messages (EIP712) are supported. Got version ${opts.version}`);
+  async signTypedData(address, msg, opts) {
+    if (opts.version && (opts.version !== 'V4' && opts.version !== 'V3')) {
+      throw new Error(
+        `Only signTypedData V3 and V4 messages (EIP712) are supported. Got version ${opts.version}`
+      );
+    }
     return this.signMessage(address, { payload: msg, protocol: 'eip712' })
   }
 
-  signMessage(address, msg) {
-    return new Promise((resolve, reject) => {
-      this._findSignerIdx(address)
-      .then((accountIdx) => {
-        let { payload, protocol } = msg;
-        // If the message is not an object we assume it is a legacy signPersonal request
-        if (!payload || !protocol) {
-          payload = msg;
-          protocol = 'signPersonal';
-        }
-        const addressIdx = this.accountIndices[accountIdx];
-        const addressParentPath = this.accountOpts[accountIdx].hdPath;
-        const req = {
-          currency: 'ETH_MSG',
-          data: {
-            protocol,
-            payload,
-            signerPath: this._getHDPathIndices(addressParentPath, addressIdx),
-          }
-        }
-        this.sdkSession.sign(req, (err, res) => {
-          if (err) {
-            return reject(new Error(err));
-          }
-          if (!this._syncCurrentWalletUID()) {
-            return reject('No active wallet.');
-          }
-          if (!res.sig) {
-            return reject(new Error('No signature returned'));
-          }
-          // Convert the `v` to a number. It should convert to 0 or 1
-          try {
-            let v = res.sig.v.toString('hex');
-            if (v.length < 2) {
-              v = `0${v}`;
-            }
-            return resolve(`0x${res.sig.r}${res.sig.s}${v}`);
-          } catch (err) {
-            return reject(new Error('Invalid signature format returned.'))
-          }
-        })
-      })
-      .catch((err) => {
-        return reject(new Error(err));
-      })
-    })
+  async signMessage (address, msg) {
+    const accountIdx = await this._findSignerIdx(address);
+    let { payload, protocol } = msg;
+    // If the message is not an object we assume it is a legacy signPersonal request
+    if (!payload || !protocol) {
+      payload = msg;
+      protocol = "signPersonal";
+    }
+    const addressIdx = this.accountIndices[accountIdx];
+    const addressParentPath = this.accountOpts[accountIdx].hdPath;
+    const req = {
+      currency: "ETH_MSG",
+      data: {
+        protocol,
+        payload,
+        signerPath: this._getHDPathIndices(addressParentPath, addressIdx),
+      },
+    };
+    const res = await this.sdkSession.sign(req);
```

Comment:
If `sign()` fails, we might want to pass the error up the call stack, so could do this: 

```suggestion
    const res = await this.sdkSession.sign(req).catch(err => { throw(err) });
```

or handle it some other way.

Reply:
The error will get thrown by `signMessage` though. Isn't what you are suggesting equivalent to what's in there now?